backpack.tf extended sorting.user.js

```diff
@@ -871,6 +874,65 @@ class</a></li>
 
 		}
 
+		var cnSort = 0;
+
+		function sortByCraftNumber() {
+			const sortCN = (a, b) => {
+				try {
+					return Number($(a)[0].outerText.split("#")[1]) - Number($(b)[0].outerText.split("#")[1]);
+				} catch (e) {
+					return parseInt(a[0].split(" ")[1]) - parseInt(b[0].split(" ")[1]);
+				}
+			}
+			let sorts = [
+				(a, b) => sortCN(a,b),
+				(a, b) => sortCN(b,a)
+			]
+			let cn = {};
+			cn["Craft Number"] = {
+				"cc": ["#DD5522"],
+				"items": []
+			};
+
+			cn["No Craft Number"] = {
+				"cc": ["#676780"],
+				"items": []
+			};
+
+			cn["Hidden Items"] = {
+				"cc": ["#676780"]
+			};
+
+			let z = $(".backpack-page .item:not(.spacer)");
+
+			cn["Hidden Items"]["items"] = $(".temp-page .item:not(.spacer)");
+			for (let p = 0; p < z.length; p++) {
+				if ($(z[p])[0].outerText.includes("#")) {
+					cn["Craft Number"]["items"].push($(z[p])[0]);
+				} else {
```

Comment:
Just this also includes crate / cases series in the craft number category. I would say it makes more sense to have an additional category for them. So you still get the sorted crates, but don't have them within other items.
```suggestion
				if ($(z[p])[0].outerText.includes("#")) {
					if($(z[p]).data("crate") == undefined) {
						cn["Craft Number"]["items"].push($(z[p])[0]);
					} else {
						cn["Crates"]["items"].push($(z[p])[0]);
					}
				} else {
```

For that of course something along the lines of 

```
			cn["Crates"] = {
				"cc": ["#676780"],
				"items": []
			};
```
needs to be inserted too at the top.